types/time/time.go

```diff
@@ -0,0 +1,49 @@
+package time
+
+import (
+	"sort"
+	"time"
+)
+
+// Now returns UTC time rounded since the zero time.
+func Now() time.Time {
+	return time.Now().Round(0).UTC()
```

Comment:
Can we clarify what we achieve with the `Round(0)` here ? It get's rid of the monotonic part according to https://golang.org/pkg/time/#Time.Round and https://golang.org/pkg/time/#hdr-Monotonic_Clocks but not 100% sure why we're doing that.

Reply:
@jaekwon Can you please provide some input here? I have imported this from your old BFT time  PR.

Reply:
Stripping it is for time equality.

If we want to compare timestamps for equality, and you have time that was generated without Round(0) and compare it against one that was, they won't be equal.  This might happen when you generate a block or vote (which generates a time locally, which without Round(0) would include the monotonic component), and then later you want to compare the timestamp there which was generated locally, against something that you encoded/decoded, using reflect.DeepEqual or assert/require.Equal.

We certainly don't need to use the monotonic part for most usages of time, because when we're setting the time of anything on the blockchain, we only care about what is encoded, nothing subjective.  So, in order to make reflect.DeepEqual and other equality methods usable easily with Tendermint types, we should strip the monotonic component when creating them, like with this Now() method (for most usages of time anyways).

I remember that I was motivated to do this because a test was failing, I forget which.

Reply:
Ok I'll add a comment to the code in https://github.com/tendermint/tendermint/pull/2343#discussion_r215717654